### PR TITLE
Collections and temporary directory parameters

### LINKED ISSUE #252: Zimit does not really support setting the `--collection` CLI argument
Hello,

I'm trying to get my hands on zimit. I've tryed a few time always ending the same way. 

here is the parameters I use:
docker run  -v /output:/output --shm-size=1gb ghcr.io/openzim/zimit zimit --url https://[Website to crowl]/ --name [name for the file to be generated] --title "Title of the zim file" --description "short description" --long-description "Long description" --collection "tag" --zim-lang en --workers 20 --waitUntil domcontentloaded --scopeType prefix

the cralw run fine until i get a message "Exiting, crawl status: done" and just after the program crashes with and "IndexError: list index out of range" message.

The full Trace reads
File "/usr/bin/zimit", line 546, in <module>
    zimiz()
File "/usr/bin/zimit", line 440, in zimit
    warc_files = list(temp_root_dir.rglog("collections/crawl-*/archive/"))[-1]
IndexError:  list index out of range

Any idea why?

Feel free to ask if you need more info

Have a nice day

### PR DESCRIPTION
**Pull request by benoit74**
# Rationale
- Fix #252 
- Add ability to dissociate output from temporary directories

# Changes
- `--collection` parameter is now used correctly to find WARC files
- more logs are displayed explaining the search for WARC files location
- new `--build` parameter (optional) to place Browsertrix files in a different directory that ZIM files 
  - ZIMs are always placed in `--output` directory
  - Browsertrix files are placed in a "temporary" subdir of `--build` (if supplied) or `--output` by default
- display `warc2zim` arguments

### CODE REVIEW

**Comment by benoit74** [comment]
@rgaudin @kelson42 I finally slightly changed my opinion compared to Thursday discussion, I fixed the `--collection` parameter logic, it was not that complex to do (and a bit sad to remove that even if not really useful) ; and it made me realized that it was not possible to set a distinct dir for WARC files location so I added this as well.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
```
Comment (rgaudin) [inline CHANGELOG.md:8]: This is not respecting the format. Use `[Unreleased]`

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 
+### Added
+
+- `--tmp` parameter (optional)
```
Comment (rgaudin) [inline CHANGELOG.md:12]: `tmp` is not clear enough about what this does. We usually call this kind of thing `build-dir` in other scrapers.

Also pleas include a brief summary of what it does here in the changelog

File: zimit.py [hunk 3]
```diff
@@ -431,14 +437,30 @@ def cleanup():
...
-    warc_files = list(temp_root_dir.rglob("collections/crawl-*/archive/"))[-1]
-    warc2zim_args.append(str(warc_files))
-
-    num_files = sum(1 for e in warc_files.iterdir())
+    if zimit_args.collection:
+        warc_files = temp_root_dir.joinpath(
+            f"collections/{zimit_args.collection}/archive/"
+        )
+    else:
+        warc_files = list(temp_root_dir.rglob("collections/crawl-*/archive/"))
```
Comment (rgaudin) [inline zimit.py:445]: This is not WARC files, it's a list of folders.

File: zimit.py [hunk 4]
```diff
@@ -431,14 +437,30 @@ def cleanup():
...
+        if len(warc_files) == 0:
+            raise RuntimeError(
+                "Failed to find directory where WARC files have been created"
+            )
+        elif len(warc_files) > 1:
+            print("Found many WARC files directories, only last one will be used")
+            for directory in warc_files:
+                print(f"- {directory}")
+        warc_files = warc_files[-1]
```
Comment (rgaudin) [inline zimit.py:454]: I don't like changing a variable's type. We use it only a couple times, maybe we can use the `[-1]` there

**Review by rgaudin — CHANGES_REQUESTED** [review]
LGTM but there are some naming to fix

**Comment by benoit74** [comment]
Thank you, I also fixed a "bug" where I did not created one temporary subdir per run, which I found problematic in terms of data segregation and very/risky for the cleanup which was deleting the whole build dir while it might have contained other files before the run.

**Pull request merged**